apis/compute/v1beta1/referencers.go

```diff
@@ -39,6 +39,17 @@ func NetworkURL() reference.ExtractValueFn {
 	}
 }
 
+// NetworkSelfLink extracts the SelfLink of a Network.
+func NetworkSelfLink() reference.ExtractValueFn {
+	return func(mg resource.Managed) string {
+		n, ok := mg.(*Network)
+		if !ok {
+			return ""
+		}
+		return n.Status.AtProvider.SelfLink
+	}
+}
```

Comment:
Isn't the above function enough for this referencer?

![image](https://user-images.githubusercontent.com/11353912/196816060-e05c727d-2633-40c7-962f-a06b624f56f1.png)

Reply:
Unfortunately `NetworkURL` function is not enough because I need a full SelfLink not a trimmed version of the link. I also can't just add a prefix to the string returned by `NetworkURL()` because the `Extract` field expects a value of type `reference.ExtractValueFn`. 

Here is an error I'm getting when I'm using `NetworkURL()` instead of `NetworkSelfLink()` 

```
1.667036804796583e+09	DEBUG	provider-gcp	Cannot create external resource	{"controller": "managed/managedzone.dns.gcp.crossplane.io", "request": "/test", "uid": "6f5ba5ef-dcd2-488f-b5bc-48e3cbad11fd", "version": "800", "external-name": "test", "error": "cannot create DNS ManagedZone: googleapi: Error 400: Invalid value for 'entity.managedZone.privateVisibilityConfig.networks[0].networkUrl': 'projects/po-test-314912/global/networks/example', invalid"}
```